### PR TITLE
feat: added timer handlers

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
We need to be able to set our timer values for both modes separately.

### Requirements
1. Default focus mode timer value **should be set to 25 minutes**
2. Default break mode timer value **should be set to 5 minutes**
3. Clicking on `Minus` icon **should decrease default timer value by 5 minutes**
4. Clicking on `Plus` icon **should increase default timer value by 5 minutes**
5. Timer value **cannot be lower than 5 minutes**

### Review

#### src/App.tsx
```diff
@@ -27,6 +27,7 @@ const Separator = styled.div`
 function App() {
 	return (
 		<Container>
+			<button>huj</button>
 			<Header />
 			<Timer />
 			<StartButton text='START FOCUS' />
```
The added button in `App.tsx` does not seem to serve any purpose related to the task requirements and should be removed.

#### src/assets/icons/Minus.tsx
```diff
@@ -3,7 +3,7 @@ import React from 'react'
 export const Minus = () => {
 	return (
 		<svg xmlns='http://www.w3.org/2000/svg' width='25' height='25' viewBox='0 0 25 25' fill='none'>
-			<path d='M5.5 12.5H19.5' stroke='#FEF2E7' stroke-linecap='round' stroke-linejoin='round' />
+			<path d='M5.5 12.5H19.5' stroke='#FEF2E7' strokeLinecap='round' strokeLinejoin='round' />
 		</svg>
 	)
 }
```
The change in `Minus.tsx` is a syntax correction and does not affect the functionality related to the task requirements.

#### src/assets/icons/Plus.tsx
```diff
@@ -3,8 +3,8 @@ import React from 'react'
 export const Plus = () => {
 	return (
 		<svg xmlns='http://www.w3.org/2000/svg' width='25' height='25' viewBox='0 0 25 25' fill='none'>
-			<path d='M12.5 5.5V19.5' stroke='#FEF2E7' stroke-linecap='round' stroke-linejoin='round' />
-			<path d='M5.5 12.5H19.5' stroke='#FEF2E7' stroke-linecap='round' stroke-linejoin='round' />
+			<path d='M12.5 5.5V19.5' stroke='#FEF2E7' strokeLinecap='round' strokeLinejoin='round' />
+			<path d='M5.5 12.5H19.5' stroke='#FEF2E7' strokeLinecap='round' strokeLinejoin='round' />
 		</svg>
 	)
 }
```
The change in `Plus.tsx` is a syntax correction and does not affect the functionality related to the task requirements.

#### src/components/Timer.tsx
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import React, { useState } from 'react'
 import styled from 'styled-components'
 import {
 	orbitAnimation1,
@@ -54,6 +54,8 @@ const TimerButton = styled.button`
 	border-radius: 80px;
 	border: 0.5px solid var(--Beige, #fef2e7);
 	background: var(--Dark, #000300);
+	cursor: pointer;
+	z-index: 100;
 `
 
 const Circle = styled.div<CircleProps>`
@@ -70,14 +72,28 @@ const Circle = styled.div<CircleProps>`
 `
 
 const Timer = () => {
+	const [focusTime, setFocusTime] = useState(25)
+	const [breakTime, setBreakTime] = useState(5)
+
+	const handleDecrease = () => {
+		setFocusTime(prev => Math.max(prev - 5, 5))
+		setBreakTime(prev => Math.max(prev - 5, 5))
+	}
+
+	const handleIncrease = () => {
+		setFocusTime(prev => prev + 5)
+		setBreakTime(prev => prev + 5)
+	}
+
 	return (
 		<TimerContainer>
-			<TimeDisplay>25:00</TimeDisplay>
+			<button onClick={handleDecrease}>ghghhg</button>
+			<TimeDisplay>{focusTime}:00</TimeDisplay>
 			<ButtonsContainer>
-				<TimerButton>
+				<TimerButton onClick={handleDecrease}>
 					<Minus />
 				</TimerButton>
-				<TimerButton>
+				<TimerButton onClick={handleIncrease}>
 					<Plus />
 				</TimerButton>
 			</ButtonsContainer>
```
The changes in `Timer.tsx` partially meet the task requirements. The default focus mode timer value is set to 25 minutes and the default break mode timer value is set to 5 minutes. The `Minus` and `Plus` buttons decrease and increase the timer values by 5 minutes respectively, and the timer value cannot be lower than 5 minutes. However, the task requires that we should be able to set timer values for both modes separately, but the current implementation changes both timer values simultaneously. The added button with the text "ghghhg" does not seem to serve any purpose related to the task requirements and should be removed.

### Suggestions
The `handleDecrease` and `handleIncrease` functions should be modified to change the timer values for the focus and break modes separately. This could be achieved by passing a parameter to these functions to indicate which mode's timer value should be changed. For example:

```javascript
const handleDecrease = (mode) => {
	if (mode === 'focus') {
		setFocusTime(prev => Math.max(prev - 5, 5))
	} else if (mode === 'break') {
		setBreakTime(prev => Math.max(prev - 5, 5))
	}
}

const handleIncrease = (mode) => {
	if (mode === 'focus') {
		setFocusTime(prev => prev + 5)
	} else if (mode === 'break') {
		setBreakTime(prev => prev + 5)
	}
}
```

The `onClick` handlers for the `Minus` and `Plus` buttons should then be updated to pass the appropriate mode to these functions:

```javascript
<TimerButton onClick={() => handleDecrease('focus')}>
	<Minus />
</TimerButton>
<TimerButton onClick={() => handleIncrease('focus')}>
	<Plus />
</TimerButton>
```

This would allow the timer values for the focus and break modes to be set separately, as required by the task.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
The goal of the task is to set timer values for both modes separately. The default focus mode timer value should be set to 25 minutes and the default break mode timer value should be set to 5 minutes. Clicking on the Minus icon should decrease the default timer value by 5 minutes, and clicking on the Plus icon should increase the default timer value by 5 minutes. The timer value cannot be lower than 5 minutes.

### Code Changes Review

#### src/assets/icons/Minus.tsx
The change in this file is related to the SVG attribute naming convention in JSX. The `stroke-linecap` and `stroke-linejoin` attributes have been changed to `strokeLinecap` and `strokeLinejoin` respectively. This change is correct as JSX uses camelCase naming convention for attributes.

#### src/assets/icons/Plus.tsx
The change in this file is similar to the change in Minus.tsx. The `stroke-linecap` and `stroke-linejoin` attributes have been changed to `strokeLinecap` and `strokeLinejoin` respectively. This change is correct as JSX uses camelCase naming convention for attributes.

#### src/components/Timer.tsx
The changes in this file are related to the task requirements. 

- The `useState` hook has been imported from React and two state variables, `focusTime` and `breakTime`, have been declared with initial values of 25 and 5 respectively. This is in line with the task requirements.

- Two new functions, `handleDecrease` and `handleIncrease`, have been added. These functions take a mode ('focus' or 'break') as a parameter and decrease or increase the corresponding timer value by 5 minutes. The decrease function also ensures that the timer value cannot be lower than 5 minutes. These changes are in line with the task requirements.

- The `onClick` event handlers of the TimerButton components have been updated to call the `handleDecrease` and `handleIncrease` functions with 'focus' as the argument. This means that the focus timer value will be updated when the Minus or Plus icon is clicked. This is in line with the task requirements.

However, there are a few issues with the changes:

1. The `breakTime` state variable is not used anywhere in the component. According to the task requirements, we need to be able to set our timer values for both modes separately. Therefore, there should be a way to switch between focus mode and break mode and update the corresponding timer value.

2. The time display only shows the focus time. It should show the current timer value depending on the mode.

3. The timer value is displayed as an integer followed by ":00". This may not be accurate when the timer value is less than 10 minutes because it will be displayed as a single digit followed by ":00". It would be better to always display the timer value as a two-digit number.

Here is an example of how these issues can be addressed:

```jsx
const Timer = () => {
    const [mode, setMode] = useState('focus');
    const [focusTime, setFocusTime] = useState(25);
    const [breakTime, setBreakTime] = useState(5);

    const handleDecrease = () => {
        if (mode === 'focus') {
            setFocusTime(prev => Math.max(prev - 5, 5));
        } else if (mode === 'break') {
            setBreakTime(prev => Math.max(prev - 5, 5));
        }
    };

    const handleIncrease = () => {
        if (mode === 'focus') {
            setFocusTime(prev => prev + 5);
        } else if (mode === 'break') {
            setBreakTime(prev => prev + 5);
        }
    };

    const currentTimerValue = mode === 'focus' ? focusTime : breakTime;

    return (
        <TimerContainer>
            <TimeDisplay>{currentTimerValue.toString().padStart(2, '0')}:00</TimeDisplay>
            <ButtonsContainer>
                <TimerButton onClick={handleDecrease}>
                    <Minus />
                </TimerButton>
                <TimerButton onClick={handleIncrease}>
                    <Plus />
                </TimerButton>
            </ButtonsContainer>
        </TimerContainer>
    );
};
```

In this updated code, a new state variable `mode` is used to keep track of the current mode. The `handleDecrease` and `handleIncrease` functions are updated to use the `mode` state variable. The `currentTimerValue` is determined based on the mode and is displayed in the TimeDisplay component. The timer value is converted to a string and padded with a leading zero if necessary using the `padStart` method.

**Review by NerdbordMentor — APPROVED** [review]
Based on the provided task requirements and the code changes, here is the code review:

1. **src/assets/icons/Minus.tsx & src/assets/icons/Plus.tsx**: The changes in these files are only related to the SVG attributes naming convention. The changes do not affect the functionality of the icons and do not directly relate to the task requirements.

2. **src/components/Timer.tsx**: This file has significant changes that cover the task requirements:

    - The default focus mode timer value is set to 25 minutes and the default break mode timer value is set to 5 minutes as required.
    - The `handleDecrease` and `handleIncrease` functions are implemented to decrease and increase the timer value by 5 minutes respectively when clicking on the Minus and Plus icons.
    - The timer value cannot be lower than 5 minutes. This is ensured by using `Math.max(prev.minutes - 5, 5)` in the `handleDecrease` function.
    - The timer values for both modes can be set separately as required.

3. **tsconfig.json**: The addition of "lib": ["dom", "es2017"] in the compilerOptions is a configuration change that does not directly relate to the task requirements.

In conclusion, the code changes cover all the task requirements. There are no apparent bugs in the provided code changes.
